kurtosis-module/impl/cl_client_network/lighthouse/lighthouse_launcher.go

```diff
@@ -0,0 +1,216 @@
+	if !found {
+		return nil, stacktrace.NewError("Expected new Lighthouse service to have port with ID '%v', but none was found", httpPortID)
+	}
+
+	restClient := cl_client_rest_client.NewCLClientRESTClient(serviceCtx.GetPrivateIPAddress(), httpPort.GetNumber())
+
+	if err := waitForAvailability(restClient); err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred waiting for the new Lighthouse node to become available")
+	}
+
+	nodeIdentity, err := restClient.GetNodeIdentity()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred getting the new Lighthouse node's identity, which is necessary to retrieve its ENR")
+	}
+
+	result := cl_client_network.NewConsensusLayerClientContext(
+		serviceCtx,
+		nodeIdentity.ENR,
+	)
+
+	return result, nil
+}
+
+func (launcher *LighthouseCLClientLauncher) getContainerConfigSupplier(
+	bootNodeEnr string,
+	elClientRpcSockets map[string]bool,
+	totalTerminalDiffulty uint32,
+) func(string, *services.SharedPath) (*services.ContainerConfig, error) {
+	return func(privateIpAddr string, sharedDir *services.SharedPath) (*services.ContainerConfig, error) {
+		configDataDirpathOnServiceSharedPath := sharedDir.GetChildPath(configDataDirpathRelToSharedDirRoot)
+
+		destConfigDataDirpathOnModule := configDataDirpathOnServiceSharedPath.GetAbsPathOnThisContainer()
+		if err := recursive_copy.Copy(launcher.configDataDirpathOnModuleContainer, destConfigDataDirpathOnModule); err != nil {
+			return nil, stacktrace.Propagate(
+				err,
+				"An error occurred copying the config data directory on the module, '%v', into the service container, '%v'",
+				launcher.configDataDirpathOnModuleContainer,
+				destConfigDataDirpathOnModule,
+			)
+		}
+
+		elClientRpcUrls := []string{}
+		for rpcSocketStr := range elClientRpcSockets {
+			rpcUrlStr := fmt.Sprintf("http://%v", rpcSocketStr)
+			elClientRpcUrls = append(elClientRpcUrls, rpcUrlStr)
+		}
+		elClientRpcUrlsStr := strings.Join(elClientRpcUrls, ",")
+
+		configDataDirpathOnService := configDataDirpathOnServiceSharedPath.GetAbsPathOnServiceContainer()
+		// NOTE: If connecting to the merge devnet remotely we DON'T want the following flags; when they're not set, the node's external IP address is auto-detected
+		//  from the peers it communicates with but when they're set they basically say "override the autodetection and
+		//  use what I specify instead." This requires having a know external IP address and port, which we definitely won't
+		//  have with a network running in Kurtosis.
+		//    "--disable-enr-auto-update",
+		//    "--enr-address=" + externalIpAddress,
+		//    fmt.Sprintf("--enr-udp-port=%v", discoveryPortNum),
+		//    fmt.Sprintf("--enr-tcp-port=%v", discoveryPortNum),
+		cmdArgs := []string{
+			"lighthouse",
+			"--debug-level=info",
+			"--datadir=" + consensusDataDirpathOnServiceContainer,
+			"--testnet-dir=" + configDataDirpathOnService,
+			"bn",
+			"--eth1",
```

Comment:
this flag is eth1. Is this correct?

Reply:
I think so - just copied from the Docker compose

Reply:
ok, good